### PR TITLE
Manage AWS Nat Gateways ec2_vpc_nat_gateway.

### PR DESCRIPTION
**Pull request by linuxdynasty**
##### Issue Type:
- New Module Pull Request
##### Plugin Name:

ec2_vpc_nat_gateway
##### Ansible Version:

```
ansible 2.1.0
```
##### Summary:

I based this module off of @Etherdaemon original pull request. 
    Manage AWS Nat Gateways.
    \* Create an AWS Nat Gateway.
    \* Delete an AWS Nat Gateway.
    \* If Nat Gateway exist in subnet and the option is passed to not create one, it will then return the Nat Gateway object.
##### Example output:

This is deleting a nat gateway.

```
***********************************
PARSED OUTPUT
{
    "changed": true,
    "invocation": {
        "module_args": {
            "allocation_id": null,
            "aws_access_key": null,
            "aws_secret_key": null,
            "client_token": null,
            "ec2_url": null,
            "eip_address": null,
            "if_exist_do_not_create": false,
            "nat_gateway_id": "nat-0b2f9f2ac3f51a653",
            "profile": null,
            "region": null,
            "release_eip": true,
            "security_token": null,
            "state": "absent",
            "subnet_id": null,
            "validate_certs": true,
            "wait": true,
            "wait_timeout": 320
        }
    },
    "result": {
        "create_time": "2016-03-05T05:19:20.282000+00:00",
        "delete_time": "2016-03-05T22:07:29.120000+00:00",
        "nat_gateway_addresses": [
            {
                "allocation_id": "eipalloc-12345",
                "network_interface_id": "eni-12345",
                "private_ip": "10.0.0.100",
                "public_ip": "52.52.52.52"
            }
        ],
        "nat_gateway_id": "nat-0b2f9f2ac3f51a653",
        "state": "deleted",
        "subnet_id": "subnet-12345",
        "vpc_id": "vpc-12345"
    }
}
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @linuxdynasty for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

**Comment by mingchuno** [comment]
```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/Users/mcor/.ansible/tmp/ansible-tmp-1458546036.5-249823338438206/ec2_vpc_nat_gateway", line 3114, in <module>
    main()
  File "/Users/mcor/.ansible/tmp/ansible-tmp-1458546036.5-249823338438206/ec2_vpc_nat_gateway", line 877, in main
    get_aws_connection_info(module, boto3=True)
  File "/Users/mcor/.ansible/tmp/ansible-tmp-1458546036.5-249823338438206/ec2_vpc_nat_gateway", line 3023, in get_aws_connection_info
    region = boto.config.get('Boto', 'aws_region')
NameError: global name 'boto' is not defined

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "ec2_vpc_nat_gateway"}, "parsed": false}
```

**Comment by linuxdynasty** [comment]
@mingchuno do you have boto and boto3 installed? My module only requires boto3 and botocore installed. From that error it seems that the function get_aws_connection_info in ec2.py is calling boto. Even though I set get_aws_connection_info(module, boto3=True) on line 877. @mingchuno please install boto as well if you haven't already. Also can you put what version of ansible you are using.

**Comment by linuxdynasty** [comment]
I'm closing this PR as I somehow included a bunch of other commits with my rebase. The new PR is  #1882. @mingchuno please checkout my new PR as I am closing this one.

**Comment by mingchuno** [comment]
@linuxdynasty yes, i have `pip install boto3`, let me try again in your new PR
